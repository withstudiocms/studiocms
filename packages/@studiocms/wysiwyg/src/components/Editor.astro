---
import "@studiocms/ui/css/colors.css";
import "grapesjs/dist/css/grapes.min.css";
import "../editorPlugins/rte/styles.css";
import "../styles/editor.css";
import { getRegistryComponents } from "studiocms:component-registry/runtime";
import type { PluginPageTypeEditorProps } from "studiocms/types";

// Get the component registry for the editor
const componentRegistry = getRegistryComponents();

interface Props extends PluginPageTypeEditorProps {}

// Extract the content from Astro props
const { content } = Astro.props as Props;
---

<div
    class="scms-grapesjs-container"
    data-component-registry={JSON.stringify(componentRegistry)}
>
    <div class="editor">
        <div id="gjs"></div>
    </div>
</div>

<textarea id="page-content" name="page-content" style="display: none;"
    >{content}</textarea
>

<script>
    import { toast } from "@studiocms/ui/components/Toast/toast.js";
    import grapesjs, { usePlugin } from "grapesjs";
    import { baseConfig } from "../common/editor-settings.js";
    import { getEditorElmData, inlineStorage } from "../common/editor-utils.js";
    import { astroComponents } from "../common/componentRegistry.js";
    import grapesBlocks from "../common/grapesBlocks/index.js";
    import {
        blocks,
        customCode,
        countdown,
        forms,
        rte,
        tabs,
        tuiImageEditor,
        typed,
    } from "../editorPlugins/index.js";

    // Get the editor options for Astro components and inline storage
    // This will be used to initialize the plugins with the correct settings
    const { astroComponentsOpts, inlineStorageOpts } = getEditorElmData(
        document,
        {
            container: ".scms-grapesjs-container",
            pageContent: "#page-content",
        },
    );

    // Initialize the GrapesJS editor with the base configuration and plugins
    // The plugins include custom components, inline storage, and various editor features
    // such as blocks, tabs, countdown, custom code, tooltips, image editor,
    // typed text, forms, rich text editor, and the StudioCMS GrapesJS plugin
    // This setup allows for a rich editing experience with various functionalities
    const editor = grapesjs.init({
        ...baseConfig,
        plugins: [
            usePlugin(astroComponents, astroComponentsOpts),
            usePlugin(inlineStorage, inlineStorageOpts),
            usePlugin(grapesBlocks),
            usePlugin(blocks),
            usePlugin(tabs),
            usePlugin(countdown),
            usePlugin(customCode),
            usePlugin(tuiImageEditor),
            usePlugin(typed),
            usePlugin(forms),
            usePlugin(rte),
        ],
    });

    // Simple warn notifier
    const origWarn = console.warn;
    console.warn = function (msg) {
        if (msg.indexOf("[undefined]") == -1) {
            toast({
                title: "Warning",
                type: "warning",
                description: msg,
                duration: 10000,
            });
        }
        origWarn(msg);
    };

    // Do stuff on load
    editor.on("load", function () {
        var $ = grapesjs.$;

        // Load and show settings and style manager
        var openTmBtn = editor.Panels.getButton("views", "open-tm");
        openTmBtn && openTmBtn.set("active", 1);
        var openSm = editor.Panels.getButton("views", "open-sm");
        openSm && openSm.set("active", 1);

        // Remove trait view
        editor.Panels.removeButton("views", "open-tm");

        // Add Settings Sector
        var traitsSector = $(
            '<div class="gjs-sm-sector no-select">' +
                '<div class="gjs-sm-sector-title"><span class="icon-settings fa fa-cog"></span> <span class="gjs-sm-sector-label">Settings</span></div>' +
                '<div class="gjs-sm-properties" style="display: none;"></div></div>',
        );
        var traitsProps = traitsSector.find(".gjs-sm-properties");
        traitsProps.append($(".gjs-traits-cs"));
        $(".gjs-sm-sectors").before(traitsSector);
        traitsSector.find(".gjs-sm-sector-title").on("click", function () {
            var traitStyle = traitsProps.get(0).style;
            var hidden = traitStyle.display == "none";
            if (hidden) {
                traitStyle.display = "block";
            } else {
                traitStyle.display = "none";
            }
        });

        // Open block manager
        var openBlocksBtn = editor.Panels.getButton("views", "open-blocks");
        openBlocksBtn && openBlocksBtn.set("active", 1);
    });
</script>
