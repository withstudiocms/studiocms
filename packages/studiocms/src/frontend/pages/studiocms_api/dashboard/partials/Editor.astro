---
export const partial = true;

import pluginsList from 'studiocms:plugins';
import { AstroError } from 'astro/errors';

type RawJsonData = {
	content: string | undefined;
	editor: string;
};

type EditorComponent = (_props?: {
	content?: string | null | undefined;
	// biome-ignore lint/suspicious/noExplicitAny: This is a valid use case for explicit any.
}) => any;

type EditorComponentItem = {
	identifier: string;
	Component: EditorComponent;
};

async function load() {
	const jsonData: RawJsonData | undefined = await Astro.request.json();

	const pageTypeComponents: EditorComponentItem[] = [];

	for (const { pageTypes } of pluginsList) {
		if (!pageTypes) continue;

		for (const { identifier, pageContentComponent } of pageTypes) {
			if (!pageContentComponent) continue;

			pageTypeComponents.push({
				identifier,
				Component: (await import(/* @vite-ignore */ pageContentComponent)).default,
			});
		}
	}

	let Editor: EditorComponentItem | undefined;

	Editor = pageTypeComponents.find((ed) => ed.identifier === jsonData?.editor);

	if (!Editor) {
		const MarkdownEditor = pageTypeComponents.find(
			(editor) => editor.identifier === 'studiocms/markdown'
		);
		if (MarkdownEditor) {
			Editor = MarkdownEditor;
		} else {
			throw new AstroError(
				`No editor found for identifier: ${jsonData?.editor}`,
				'StudioCMSDashboardPartialEditorError - No editor found'
			);
		}
	}

	if (jsonData?.content) {
		return {
			content: jsonData.content,
			ActiveEditor: Editor.Component,
		};
	}

	return {
		content: 'No content to display',
		ActiveEditor: Editor.Component,
	};
}

const { content, ActiveEditor } = await load();
---

<ActiveEditor content={content} />
