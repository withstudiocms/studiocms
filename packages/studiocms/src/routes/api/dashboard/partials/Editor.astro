---
export const partial = true;
import pluginsList from 'studiocms:plugins';

const pageTypeComponents = async () => {
	const pageTypeComponents: {
		identifier: string;
		// biome-ignore lint/suspicious/noExplicitAny: <explanation>
		Component: (Props?: { content?: string | null | undefined }) => any;
	}[] = [];

	for (const { pageTypes } of pluginsList) {
		if (!pageTypes) {
			return pageTypeComponents;
		}

		for (const { identifier, pageContentComponent } of pageTypes) {
			if (!pageContentComponent) {
				continue;
			}

			pageTypeComponents.push({
				identifier,
				Component: (await import(/* @vite-ignore */ pageContentComponent)).default,
			});
		}

		return pageTypeComponents;
	}

	return pageTypeComponents;
};

async function setContent() {
	const jsonData: { content: string | undefined; editor: string } | undefined =
		await Astro.request.json();

	const editors = await pageTypeComponents();

	let currentEditor = editors.find((ed) => {
		if (ed.identifier === jsonData?.editor) {
			return ed;
		}
	});

	if (!currentEditor) {
		// biome-ignore lint/style/noNonNullAssertion: <explanation>
		currentEditor = editors.find((editor) => editor.identifier === 'studiocms/markdown')!;
	}

	if (jsonData?.content) {
		return { content: jsonData.content, ActiveEditor: currentEditor.Component };
	}

	return { content: 'No content to display', ActiveEditor: currentEditor.Component };
}

const { content, ActiveEditor } = await setContent();
---
<ActiveEditor content={content} />