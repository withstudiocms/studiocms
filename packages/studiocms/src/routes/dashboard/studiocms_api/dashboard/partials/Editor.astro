---
import { pageTypeComponents } from '../../../../../components/dashboard/islands/content-mgmt/shared.js';

export const partial = true;

async function setContent() {
	const jsonData: { content: string | undefined; editor: string } | undefined =
		await Astro.request.json();

	const editors = await pageTypeComponents();

	let currentEditor = editors.find((ed) => {
		if (ed.identifier === jsonData?.editor) {
			return ed;
		}
	});

	if (!currentEditor) {
		// biome-ignore lint/style/noNonNullAssertion: <explanation>
		currentEditor = editors.find((editor) => editor.identifier === 'studiocms/markdown')!;
	}

	if (jsonData?.content) {
		return { content: jsonData.content, ActiveEditor: currentEditor.Component };
	}

	return { content: 'No content to display', ActiveEditor: currentEditor.Component };
}

const { content, ActiveEditor } = await setContent();
---
<ActiveEditor content={content} />