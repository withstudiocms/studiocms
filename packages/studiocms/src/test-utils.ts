import type { AstroIntegration, AstroIntegrationLogger } from 'astro';
import type { StudioCMSPlugin } from './plugins.js';
import type {
	SCMSAuthServiceFnOpts,
	SCMSDashboardFnOpts,
	SCMSFrontendFnOpts,
	SCMSImageServiceFnOpts,
	SCMSRenderingFnOpts,
	SCMSSiteMapFnOpts,
} from './schemas/index.js';

type HookRun<T> = { hasHook: boolean; hookResults: T };
export interface PluginHookResults {
	astroConfig: HookRun<{ integrations: AstroIntegration[] }>;
	studiocmsConfig: HookRun<{
		authService: Partial<SCMSAuthServiceFnOpts>;
		dashboard: Partial<SCMSDashboardFnOpts>;
		frontend: Partial<SCMSFrontendFnOpts>;
		imageService: Partial<SCMSImageServiceFnOpts>;
		rendering: Partial<SCMSRenderingFnOpts>;
		sitemap: Partial<SCMSSiteMapFnOpts>;
	}>;
}
/**
 * Utility class for testing StudioCMS plugins by simulating hook execution and collecting results.
 *
 * The `StudioCMSPluginTester` provides methods to:
 * - Create mock loggers for use in plugin hooks.
 * - Execute the `studiocms:astro:config` and `studiocms:config:setup` hooks, if present, on a given plugin.
 * - Collect and return integration and configuration data set by these hooks.
 * - Retrieve basic plugin metadata.
 *
 * @remarks
 * This class is intended for use in test environments to facilitate inspection and validation of plugin behavior.
 *
 * @example
 * ```typescript
 * const tester = new StudioCMSPluginTester(myPlugin);
 * const info = tester.getPluginInfo();
 * const hookResults = await tester.getHookResults();
 * ```
 */
export class StudioCMSPluginTester {
	private plugin: StudioCMSPlugin;
	private readonly injectedLogger?: AstroIntegrationLogger;

	constructor(plugin: StudioCMSPlugin, logger?: AstroIntegrationLogger) {
		this.plugin = plugin;
		this.injectedLogger = logger;
	}

	/**
	 * Creates a mock logger object for testing purposes.
	 *
	 * The returned logger implements the `AstroIntegrationLogger` interface and provides
	 * stubbed logging methods (`info`, `warn`, `error`, `debug`) that delegate to the corresponding
	 * `console` methods. It also includes a `fork` method, which returns a new mock logger
	 * instance with an attached label for testing logger forking behavior.
	 *
	 * @returns {AstroIntegrationLogger} A mock logger instance suitable for use in tests.
	 */
	private createMockLogger(): AstroIntegrationLogger {
		if (this.injectedLogger) return this.injectedLogger;
		// biome-ignore lint/suspicious/noExplicitAny: this is fine
		const logger: Record<string, any> = {
			info: console.log,
			warn: console.warn,
			error: console.error,
			debug: console.debug,
			fork: (label: string) => {
				// Each fork returns a new mock logger with the label attached for testing
				const forked = this.createMockLogger();
				forked.label = label;
				return forked;
			},
		};
		return logger as unknown as AstroIntegrationLogger;
	}

	/**
	 * Executes the 'studiocms:astro:config' hook if it exists on the plugin,
	 * providing a mock logger and a method to collect Astro integrations.
	 *
	 * @returns A promise that resolves to an object containing the collected integrations.
	 *
	 * @remarks
	 * This method checks if the plugin defines a 'studiocms:astro:config' hook as a function.
	 * If so, it invokes the hook with a mock logger and an `addIntegrations` callback,
	 * which accumulates integrations into an array. The collected integrations are then
	 * returned in an object.
	 */
	private async runAstroConfigHook() {
		const integrations: AstroIntegration[] = [];
		if (
			this.plugin.hooks['studiocms:astro:config'] &&
			typeof this.plugin.hooks['studiocms:astro:config'] === 'function'
		) {
			await this.plugin.hooks['studiocms:astro:config']({
				logger: this.createMockLogger(),
				addIntegrations: (newIntegrations) => {
					const toAdd = Array.isArray(newIntegrations) ? newIntegrations : [newIntegrations];
					integrations.push(...toAdd);
				},
			});
		}
		return { integrations };
	}

	/**
	 * Executes the `studiocms:config:setup` hook on the current plugin, if available,
	 * and collects configuration options set by the hook into partial option objects.
	 *
	 * This method mocks the hook's context by providing setter functions for various
	 * StudioCMS configuration aspects, such as authentication, dashboard, frontend,
	 * image service, rendering, and sitemap. Each setter captures the values provided
	 * by the hook and stores them in corresponding partial option objects.
	 *
	 * @returns A promise that resolves to an object containing the collected configuration
	 *          options for authentication service, dashboard, frontend, image service,
	 *          rendering, and sitemap.
	 *
	 * @remarks
	 * This utility is primarily intended for testing purposes, allowing inspection of
	 * configuration values set by a plugin's `studiocms:config:setup` hook.
	 */
	private async runStudioCMSConfigHook() {
		const authService: Partial<SCMSAuthServiceFnOpts> = {};
		const dashboard: Partial<SCMSDashboardFnOpts> = {};
		const frontend: Partial<SCMSFrontendFnOpts> = {};
		const imageService: Partial<SCMSImageServiceFnOpts> = {};
		const rendering: Partial<SCMSRenderingFnOpts> = {};
		const sitemap: Partial<SCMSSiteMapFnOpts> = {};

		if (
			this.plugin.hooks['studiocms:config:setup'] &&
			typeof this.plugin.hooks['studiocms:config:setup'] === 'function'
		) {
			await this.plugin.hooks['studiocms:config:setup']({
				logger: this.createMockLogger(),
				setAuthService: ({ oAuthProvider }) => {
					if (oAuthProvider) {
						authService.oAuthProvider = oAuthProvider;
					}
				},
				setDashboard: ({ dashboardGridItems, dashboardPages }) => {
					if (dashboardGridItems) {
						dashboard.dashboardGridItems = dashboardGridItems;
					}
					if (dashboardPages) {
						dashboard.dashboardPages = dashboardPages;
					}
				},
				setFrontend: ({ frontendNavigationLinks }) => {
					if (frontendNavigationLinks) {
						frontend.frontendNavigationLinks = frontendNavigationLinks;
					}
				},
				setImageService: ({ imageService: imgService }) => {
					if (imgService) {
						imageService.imageService = imgService;
					}
				},
				setRendering: ({ pageTypes }) => {
					if (pageTypes) {
						rendering.pageTypes = pageTypes;
					}
				},
				setSitemap: ({ sitemaps, triggerSitemap }) => {
					if (sitemaps) {
						sitemap.sitemaps = sitemaps;
					}
					sitemap.triggerSitemap = triggerSitemap;
				},
			});
		}

		return {
			authService,
			dashboard,
			frontend,
			imageService,
			rendering,
			sitemap,
		};
	}

	/**
	 * Retrieves information about the current plugin.
	 *
	 * @returns An object containing the plugin's identifier, name, minimum required StudioCMS version, and dependencies.
	 */
	public getPluginInfo() {
		return {
			identifier: this.plugin.identifier,
			name: this.plugin.name,
			studiocmsMinimumVersion: this.plugin.studiocmsMinimumVersion,
			requires: this.plugin.requires,
		};
	}

	/**
	 * Asynchronously retrieves the results of configured plugin hooks.
	 *
	 * @returns An object containing the presence and results of the 'studiocms:astro:config' and 'studiocms:config:setup' hooks.
	 * - `astroConfig`: Indicates if the 'studiocms:astro:config' hook exists and provides its execution results.
	 * - `studiocmsConfig`: Indicates if the 'studiocms:config:setup' hook exists and provides its execution results.
	 */
	public async getHookResults(): Promise<PluginHookResults> {
		return {
			astroConfig: {
				hasHook: !!this.plugin.hooks['studiocms:astro:config'],
				hookResults: await this.runAstroConfigHook(),
			},
			studiocmsConfig: {
				hasHook: !!this.plugin.hooks['studiocms:config:setup'],
				hookResults: await this.runStudioCMSConfigHook(),
			},
		};
	}
}
