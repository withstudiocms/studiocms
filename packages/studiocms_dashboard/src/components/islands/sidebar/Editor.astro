---
import { getUserData } from 'studiocms:auth/lib/user';
import { verifyUserPermissionLevel } from 'studiocms:auth/lib/user';
import SidebarLink from '../../SidebarLink.astro';
import { getSidebarLinks } from '../../sidebarConfig';

const lang = 'en-us';
const sidebar = getSidebarLinks(lang);

const data = await getUserData(Astro);

const isEditor = await verifyUserPermissionLevel(data, 'editor');

Astro.response.headers.set('Cache-Control', 'max-age=604800');
---
{ isEditor && sidebar.editorLinks.map(({ href, icon, title }) => (
    <SidebarLink {icon} {href}>{title}</SidebarLink>
))}