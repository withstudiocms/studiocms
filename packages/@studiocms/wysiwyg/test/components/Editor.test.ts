/// <reference types="astro/client" />
import { describe, expect, vi } from 'vitest';
import Editor from '../../src/components/Editor.astro';
import { test } from '../fixtures/AstroContainer';
import { MockAstroLocalsWithCSRF } from '../test-utils';

// Mock the component registry
vi.mock('studiocms:component-registry/runtime', () => ({
	getRegistryComponents: vi.fn(() => [
		{
			name: 'Text',
			safeName: 'text',
			props: [
				{
					name: 'class',
					type: 'string',
					optional: true,
					description: 'CSS class for the text component',
				},
			],
		},
		{
			name: 'Heading',
			safeName: 'heading',
			props: [
				{
					name: 'class',
					type: 'string',
					optional: true,
					description: 'CSS class for the heading component',
				},
				{
					name: 'level',
					type: 'number',
					optional: true,
					defaultValue: '1',
					description: 'Heading level (1-6)',
				},
			],
		},
	]),
}));

// No need to mock Astro - we'll use Container API to pass locals directly

describe('WYSIWYG Editor Component', () => {
	test('renders editor with basic content', async ({ renderComponent }) => {
		const result = await renderComponent(Editor, 'Editor', {
			props: {
				content: '<p>Hello World</p>',
				id: 'test-page-id',
			},
		});
		expect(result).toMatchSnapshot();
	});

	test('renders editor without page id', async ({ renderComponent }) => {
		const result = await renderComponent(Editor, 'Editor', {
			props: {
				content: '<h1>Title</h1>',
			},
		});
		expect(result).toMatchSnapshot();
	});

	test('renders editor with empty content', async ({ renderComponent }) => {
		const result = await renderComponent(Editor, 'Editor', {
			props: {
				content: '',
				id: 'empty-page',
			},
		});
		expect(result).toMatchSnapshot();
	});

	test('renders editor with complex HTML content', async ({ renderComponent }) => {
		const complexContent = `
			<div class="container">
				<h1>Title</h1>
				<p>Content with <strong>bold</strong> text</p>
				<ul>
					<li>Item 1</li>
					<li>Item 2</li>
				</ul>
			</div>
		`;

		const result = await renderComponent(Editor, 'Editor', {
			props: {
				content: complexContent,
				id: 'complex-page',
			},
		});
		expect(result).toMatchSnapshot();
	});

	test('renders editor with CSRF token', async ({ renderComponent }) => {
		const result = await renderComponent(Editor, 'Editor', {
			props: {
				content: '<p>Test</p>',
				id: 'csrf-test',
			},
			locals: MockAstroLocalsWithCSRF('mock-csrf-token'),
		});
		expect(result).toMatchSnapshot();
	});
});
