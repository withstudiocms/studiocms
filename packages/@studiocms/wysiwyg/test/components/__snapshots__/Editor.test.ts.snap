// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`WYSIWYG Editor Component > renders editor with CSRF token 1`] = `"<div class="scms-grapesjs-container" data-page-id="csrf-test" data-component-registry="[{&#34;name&#34;:&#34;Text&#34;,&#34;safeName&#34;:&#34;text&#34;,&#34;props&#34;:[{&#34;name&#34;:&#34;class&#34;,&#34;type&#34;:&#34;string&#34;,&#34;optional&#34;:true,&#34;description&#34;:&#34;CSS class for the text component&#34;}]},{&#34;name&#34;:&#34;Heading&#34;,&#34;safeName&#34;:&#34;heading&#34;,&#34;props&#34;:[{&#34;name&#34;:&#34;class&#34;,&#34;type&#34;:&#34;string&#34;,&#34;optional&#34;:true,&#34;description&#34;:&#34;CSS class for the heading component&#34;},{&#34;name&#34;:&#34;level&#34;,&#34;type&#34;:&#34;number&#34;,&#34;optional&#34;:true,&#34;defaultValue&#34;:&#34;1&#34;,&#34;description&#34;:&#34;Heading level (1-6)&#34;}]}]" data-editor-csrf-token="mock-csrf-token"> <div class="editor"> <div id="gjs"></div> </div> </div> <textarea id="page-content" name="page-content" style="display: none;">&lt;p&gt;Test&lt;/p&gt;</textarea> <script type="module" src="/mock/path/Editor.astro?astro&type=script&index=0&lang.ts"></script>"`;

exports[`WYSIWYG Editor Component > renders editor with basic content 1`] = `"<div class="scms-grapesjs-container" data-page-id="test-page-id" data-component-registry="[{&#34;name&#34;:&#34;Text&#34;,&#34;safeName&#34;:&#34;text&#34;,&#34;props&#34;:[{&#34;name&#34;:&#34;class&#34;,&#34;type&#34;:&#34;string&#34;,&#34;optional&#34;:true,&#34;description&#34;:&#34;CSS class for the text component&#34;}]},{&#34;name&#34;:&#34;Heading&#34;,&#34;safeName&#34;:&#34;heading&#34;,&#34;props&#34;:[{&#34;name&#34;:&#34;class&#34;,&#34;type&#34;:&#34;string&#34;,&#34;optional&#34;:true,&#34;description&#34;:&#34;CSS class for the heading component&#34;},{&#34;name&#34;:&#34;level&#34;,&#34;type&#34;:&#34;number&#34;,&#34;optional&#34;:true,&#34;defaultValue&#34;:&#34;1&#34;,&#34;description&#34;:&#34;Heading level (1-6)&#34;}]}]"> <div class="editor"> <div id="gjs"></div> </div> </div> <textarea id="page-content" name="page-content" style="display: none;">&lt;p&gt;Hello World&lt;/p&gt;</textarea> <script type="module" src="/mock/path/Editor.astro?astro&type=script&index=0&lang.ts"></script>"`;

exports[`WYSIWYG Editor Component > renders editor with complex HTML content 1`] = `
"<div class="scms-grapesjs-container" data-page-id="complex-page" data-component-registry="[{&#34;name&#34;:&#34;Text&#34;,&#34;safeName&#34;:&#34;text&#34;,&#34;props&#34;:[{&#34;name&#34;:&#34;class&#34;,&#34;type&#34;:&#34;string&#34;,&#34;optional&#34;:true,&#34;description&#34;:&#34;CSS class for the text component&#34;}]},{&#34;name&#34;:&#34;Heading&#34;,&#34;safeName&#34;:&#34;heading&#34;,&#34;props&#34;:[{&#34;name&#34;:&#34;class&#34;,&#34;type&#34;:&#34;string&#34;,&#34;optional&#34;:true,&#34;description&#34;:&#34;CSS class for the heading component&#34;},{&#34;name&#34;:&#34;level&#34;,&#34;type&#34;:&#34;number&#34;,&#34;optional&#34;:true,&#34;defaultValue&#34;:&#34;1&#34;,&#34;description&#34;:&#34;Heading level (1-6)&#34;}]}]"> <div class="editor"> <div id="gjs"></div> </div> </div> <textarea id="page-content" name="page-content" style="display: none;">
			&lt;div class=&quot;container&quot;&gt;
				&lt;h1&gt;Title&lt;/h1&gt;
				&lt;p&gt;Content with &lt;strong&gt;bold&lt;/strong&gt; text&lt;/p&gt;
				&lt;ul&gt;
					&lt;li&gt;Item 1&lt;/li&gt;
					&lt;li&gt;Item 2&lt;/li&gt;
				&lt;/ul&gt;
			&lt;/div&gt;
		</textarea> <script type="module" src="/mock/path/Editor.astro?astro&type=script&index=0&lang.ts"></script>"
`;

exports[`WYSIWYG Editor Component > renders editor with empty content 1`] = `"<div class="scms-grapesjs-container" data-page-id="empty-page" data-component-registry="[{&#34;name&#34;:&#34;Text&#34;,&#34;safeName&#34;:&#34;text&#34;,&#34;props&#34;:[{&#34;name&#34;:&#34;class&#34;,&#34;type&#34;:&#34;string&#34;,&#34;optional&#34;:true,&#34;description&#34;:&#34;CSS class for the text component&#34;}]},{&#34;name&#34;:&#34;Heading&#34;,&#34;safeName&#34;:&#34;heading&#34;,&#34;props&#34;:[{&#34;name&#34;:&#34;class&#34;,&#34;type&#34;:&#34;string&#34;,&#34;optional&#34;:true,&#34;description&#34;:&#34;CSS class for the heading component&#34;},{&#34;name&#34;:&#34;level&#34;,&#34;type&#34;:&#34;number&#34;,&#34;optional&#34;:true,&#34;defaultValue&#34;:&#34;1&#34;,&#34;description&#34;:&#34;Heading level (1-6)&#34;}]}]"> <div class="editor"> <div id="gjs"></div> </div> </div> <textarea id="page-content" name="page-content" style="display: none;"></textarea> <script type="module" src="/mock/path/Editor.astro?astro&type=script&index=0&lang.ts"></script>"`;

exports[`WYSIWYG Editor Component > renders editor without page id 1`] = `"<div class="scms-grapesjs-container" data-page-id data-component-registry="[{&#34;name&#34;:&#34;Text&#34;,&#34;safeName&#34;:&#34;text&#34;,&#34;props&#34;:[{&#34;name&#34;:&#34;class&#34;,&#34;type&#34;:&#34;string&#34;,&#34;optional&#34;:true,&#34;description&#34;:&#34;CSS class for the text component&#34;}]},{&#34;name&#34;:&#34;Heading&#34;,&#34;safeName&#34;:&#34;heading&#34;,&#34;props&#34;:[{&#34;name&#34;:&#34;class&#34;,&#34;type&#34;:&#34;string&#34;,&#34;optional&#34;:true,&#34;description&#34;:&#34;CSS class for the heading component&#34;},{&#34;name&#34;:&#34;level&#34;,&#34;type&#34;:&#34;number&#34;,&#34;optional&#34;:true,&#34;defaultValue&#34;:&#34;1&#34;,&#34;description&#34;:&#34;Heading level (1-6)&#34;}]}]"> <div class="editor"> <div id="gjs"></div> </div> </div> <textarea id="page-content" name="page-content" style="display: none;">&lt;h1&gt;Title&lt;/h1&gt;</textarea> <script type="module" src="/mock/path/Editor.astro?astro&type=script&index=0&lang.ts"></script>"`;
